x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -259,6 +271,11 @@ public Builder addApplicationPrivilege(ApplicationPrivilege privilege, Set<Strin
             return this;
         }
 
+        public Builder workflows(Set<String> workflowNames) {
+            this.workflowsRestriction = WorkflowsRestriction.resolve(workflowNames);
```

Comment:
We don't need to resolve Workflow again here. WorkflowsRestriction can really just be a StringMatcher internally, something like
```
WorkflowsRestriction(Set<String> names) {
    this.names = names;
    this.predicate = StringMatcher.of(names);
}
```